source/solvers/fluid_dynamics_matrix_free.cc

```diff
@@ -2503,6 +2491,12 @@ FluidDynamicsMatrixFree<dim>::assemble_system_rhs()
 {
   TimerOutput::Scope t(this->computing_timer, "Assemble RHS");
 
+  // Update the precomputed values needed for the evaluation of the residual.
+  // This is needed due to the alpha procedure of the non linear solver.
```

Comment:
```suggestion
  // This is needed, otherwise the line-search mechanism used in the Newton method might fail even though the Newton step should have been accepted due to a wrong evaluation of the residual and, consequently, a wrong evaluation of the step length.
```